src/sagemaker_training/mpi.py

```diff
@@ -334,6 +336,27 @@ def _create_command(self):
         for name in self._env_vars:
             command.extend(["-x", name])
 
+        def is_smddp_installed():
+            try:
+                import smdistributed.dataparallel.torch.torch_smddp
+                output = subprocess.run(
+                    ["which", "smddpmprun"],
+                    capture_output=True,
+                    text=True,
+                    check=True,
+                )
+                return output.stdout != ''
+            except (ImportError, subprocess.CalledProcessError):
+                return False
+
+        env = environment.Environment()
+        if env.is_modelparallel_enabled:
+            mp_parameters = json.loads(os.environ.get(params.SM_HP_MP_PARAMETERS))
+            ddp_dist_backend = mp_parameters.get("ddp_dist_backend", "auto")
+            if ddp_dist_backend == "auto":
+                if is_smddp_installed():
```

Comment:
Where is this binary `smddmprun` coming from? From dataparallel or modelparallel ?

Reply:
It is from dataparallel, and will be installed to DLC along with other dataparallel components.